core/generator/Generator.cc

```diff
@@ -125,8 +125,11 @@ bool Generator::Generate(const FileDescriptor *_file,
 #include <memory>
 
 #include <gz/msgs/Export.hh>
+#include <gz/utils/SuppressWarning.hh>
 
+GZ_UTILS_WARN_IGNORE__DEPRECATED_DECLARATION
```

Comment:
This works great! 

I was trying to understand how this prevents warnings in gz-msgs, but still gives warnings in downstream libraries when you try to use the `LOCAL2` enum, which is what we want. The warning this prevents is the warning emitted by just including `msgs/details/spherical_coordinates.pb.h` because that file contains
```
constexpr SphericalCoordinatesType SphericalCoordinatesType_MAX = LOCAL2;
```
which is a direct use of the `LOCAL2` enum.

when downstream libraries include `msgs/spherical_coordinates.pb.h`, they won't get a warning from that. But if they use `LOCAL2` directly, a warning is emitted.